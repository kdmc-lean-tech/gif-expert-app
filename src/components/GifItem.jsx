import React from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

const GifItem = ({ gif }) => {
  const { title, url } = gif;
  return (
    <Card className='card animate__animated animate__bounce'>
      <Card.Img variant="top" src={ url }  style={{ width: '100%' }} id='card-image'/>
      <Card.Body>
        <Card.Title id='card-title'>{ title }</Card.Title>
      </Card.Body>
    </Card>
  );
}

GifItem.propTypes = {
  gif: PropTypes.object.isRequired
};

export default GifItem;
