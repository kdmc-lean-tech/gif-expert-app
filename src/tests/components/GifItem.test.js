import { shallow } from 'enzyme';
import GifItem from '../../components/GifItem';

describe('GifItem Component Test', () => {
  const urlGif = 'https://media4.giphy.com/media/1Ot4mnOOXJawg/giphy.gif?cid=a91e1fa0y7j2gknh42f60yu2uhlalpxf4ckzn1vodn7axxdi&rid=giphy.gif';
  test('should render GifItem component', () => {
    const gif = { title: 'Kevin', url: urlGif };
    const wrapper = shallow(<GifItem gif={ gif } />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render property title', () => {
    const gif = { title: 'Kevin', url: urlGif };
    const wrapper = shallow(<GifItem gif={ gif } />);
    const textCardValue = wrapper.find('#card-title').text();
    expect(textCardValue).toEqual('Kevin');
  });
  test('should render property url', () => {
    const gif = { title: 'Kevin', url: urlGif };
    const wrapper = shallow(<GifItem gif={ gif } />);
    const urlCardImage = wrapper.find('#card-image').prop('src');
    expect(urlCardImage).toEqual(urlGif);
  });
});
