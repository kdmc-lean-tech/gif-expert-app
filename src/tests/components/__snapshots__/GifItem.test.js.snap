// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Component Test should render GifItem component 1`] = `
<Card
  body={false}
  className="card animate__animated animate__bounce"
>
  <CardImg
    id="card-image"
    src="https://media4.giphy.com/media/1Ot4mnOOXJawg/giphy.gif?cid=a91e1fa0y7j2gknh42f60yu2uhlalpxf4ckzn1vodn7axxdi&rid=giphy.gif"
    style={
      Object {
        "width": "100%",
      }
    }
    variant="top"
  />
  <CardBody>
    <CardTitle
      id="card-title"
    >
      Kevin
    </CardTitle>
  </CardBody>
</Card>
`;
